### PR TITLE
feat: introduce sorting to catalogs page

### LINKED ISSUE #1355: Introduce Sorting of Cards in Meshery Catalog
Introduce sort by option as a dropdown in Meshery Catalog, which arranges the cards alphabetically and on the basis of time of update, both in ascending and descending manner.  

For reference take a look at [meshery.layer5.io/catalog](https://meshery.layer5.io/catalog)

### Screenshots / Mockups

---
<h4>Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources</h4>

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery Website](https://github.com/meshery/meshery.io#contributing-to-the-mesheryio-website)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by Lucif3r-in**
**Description**

This PR fixes #1355 
Previous related PR #1480 
**Notes for Reviewers**
Introduced sorting for catalogs page

**[Signed commits](https://docs.meshery.io/project/contributing#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by Lucif3r-in** [comment]
@iArchitSharma @abhijeetgauravm Check this PR it has all changes you suggested

**Review by abhijeetgauravm — COMMENTED** [review]
Just for maintaining the consistency can we do either small letter one items first before capital letter one items or after the capital letter one items??

As we can see here its not maintaining the consistency here?
Here small letter one items are before capital letter one items
image

But here capital letter one items are before small letter one items. Can we make it consistent everywhere like whether before or after ??
image

image

**Comment by Lucif3r-in** [comment]
Sort feature here is converting all to lowercase then following the sort procedure. For the second photo, it checks that the first letter is same then converts the second letter to lowercase and compares it. Do we need to change this? @abhijeetgauravm

**Comment by abhijeetgauravm** [comment]
> Sort feature here is converting all to lowercase then following the sort procedure. For the second photo it converts all to lowercase, then it checks that the first letter is same then compares the second letter and arranges the cards according to it. Do we need to change this? @abhijeetgauravm . If so, then I think we have to do ASCII comparison🤔, will do as you say

No sweat @Lucif3r-in great work so far, i think its good in this case too but let's get feedback of others on this too.

**Review by abhijeetgauravm — CHANGES_REQUESTED** [review]
Can we have proper margin here as well, as we have in case of All and deployment. As we can see the item is touching to the technologies part on the left in the below cases here!
image
image
image

**Comment by Lucif3r-in** [comment]
> Can we have proper margin here as well, as we have in case of All and deployment. As we can see the item is touching to the technologies part on the left in the below cases here! image image image

I can't find any breakpoint related code in `index.html` that defines the margin. Can you please point where the code for the page is? @abhijeetgauravm

**Review by abhijeetgauravm — COMMENTED** [review]
Looks good in preview now!

**Comment by abhijeetgauravm** [comment]
Hey @Lucif3r-in Let's discuss this on website call tomorrow at 5:30 PM IST / 7 AM Central time. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit#heading=h.lohhtewfwima).

**Review by iArchitSharma — APPROVED** [review]
@Lucif3r-in It looks really good now

**Comment by Lucif3r-in** [comment]
> Hey @Lucif3r-in Let's discuss this on website call tomorrow at 5:30 PM IST / 7 AM Central time. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit#heading=h.lohhtewfwima).

Cool did that😊

**Comment by Lucif3r-in** [comment]
@ritiksaxena124 I have done the suggested changes.
image

**Comment by Lucif3r-in** [comment]
Done @ritiksaxena124 🚀🚀

**Comment by leecalcote** [comment]
merge conflict…

**Comment by Lucif3r-in** [comment]
Done @leecalcote 🚀🚀

**Comment by abhijeetgauravm** [comment]
> Done @leecalcote 🚀🚀

@vishalvivekm @sudhanshutech @ritiksaxena124

**Comment by leecalcote** [comment]
@Lucif3r-in this is a good feature to have.

**Comment by leecalcote** [comment]
@Lucif3r-in do we have an open issue on search for the catalog? If not, do you mind opening one?

We can use https://lunrjs.com or similar.

**Comment by Lucif3r-in** [comment]
> @Lucif3r-in do we have an open issue on search for the catalog? If not, do you mind opening one?
> 
> We can use https://lunrjs.com or similar.

Ya I was just about to write the same😅. Will surely open one issue about search

**Comment by Lucif3r-in** [comment]
Can you please review this early as I have a related issue #1488 on which I am working on.

File: Gemfile.lock [hunk 1]
```diff
@@ -87,6 +86,7 @@ GEM
 PLATFORMS
   arm64-darwin-21
   ruby
+  x64-mingw-ucrt
```
Comment (sudhanshutech) [inline Gemfile.lock:89]: @Lucif3r-in why this gemfile change?

**Comment by leecalcote** [comment]
@Lucif3r-in what's the meaning behind your username? Is your intention to offend?

**Comment by Lucif3r-in** [comment]
> @Lucif3r-in what's the meaning behind your username? Is your intention to offend?

Offend? I didn't get you 😅
I just had watched the Lucifer series and there were already usernames for simple Lucifer so...

**Comment by sudhanshutech** [comment]
@Lucif3r-in merge conflict

**Comment by Lucif3r-in** [comment]
@sudhanshutech can you please check now😊

**Comment by sudhanshutech** [comment]
@Lucif3r-in the gemfile manages the dependencies of project we can't delete the file. You need to restage the gemfile changes from your pr(it could not be deleted, not change unless change in any dependencies)

**Comment by Lucif3r-in** [comment]
Can anyone check and tell why am I getting a diff even if there is no change in Gemfile.lock

**Comment by iArchitSharma** [comment]
@Lucif3r-in that's a common issue for all, you don't have to commit and push Gemfile.lock file

**Comment by Lucif3r-in** [comment]
> @Lucif3r-in that's a common issue for all, you don't have to commit and push Gemfile.lock file

Now is the PR ok? or I have to do something else

**Comment by iArchitSharma** [comment]
Remove the Gemfile.lock from PR

**Comment by Lucif3r-in** [comment]
I am a little confused with this rebasing and all, I will make a new PR

**Pull request closed without merging**